app/controllers/booking_requests_controller.rb

```diff
@@ -14,12 +14,12 @@ def step_two
   def complete
     @booking_request = BookingRequestForm.new(location_id, booking_request_params)
 
-    if @booking_request.step_two_valid?
+    if @booking_request.step_two_valid_excluding_eligibility?
+      return redirect_to booking_request_ineligible_location_path(id: location_id) unless @booking_request.eligible?
+
       BookingRequests.create(@booking_request)
       redirect_to booking_request_completed_location_path(id: location_id)
     else
-      return redirect_to booking_request_ineligible_location_path(id: location_id) unless @booking_request.eligible?
-
       render :step_two
     end
```

Comment:
Don't feel like this change improved readability.

Have you consider:
```
if @booking_request.step_two_valid? && @booking_request.eligible?
  # do A
elsif @booking_request.step_two_valid? # i.e. ineligible
  # do B
else
  # do C
end
```

The reason I suggest this is that step 2 is complete (i.e. valid) regardless of eligibility criteria.